src/vs/workbench/contrib/debug/browser/debugAdapterManager.ts

```diff
@@ -49,12 +52,15 @@ export class AdapterManager implements IAdapterManager {
 		@IExtensionService private readonly extensionService: IExtensionService,
 		@IContextKeyService contextKeyService: IContextKeyService,
 		@IModeService private readonly modeService: IModeService,
-		@IDialogService private readonly dialogService: IDialogService
+		@IDialogService private readonly dialogService: IDialogService,
+		@IStorageService private readonly storageService: IStorageService
 	) {
 		this.adapterDescriptorFactories = [];
 		this.debuggers = [];
 		this.registerListeners();
+		const debuggersAvailable = this.storageService.getBoolean(DEBUGGERS_AVAILABLE_KEY, StorageScope.WORKSPACE, false);
```

Comment:
This is up to you, but would it make sense for the default value if not set to be "true"? In the most common case (desktop + local), I would expect debuggers to be available, so should we default to that?

Reply:
Good point, however for now I will be more comfortable with going with `false` because in theory we are not guaranteed that the `registerDebugAdapterFactory` will be called (and that is the method were we actually update this). In practice this method will get called in all cases. However I still think that this way it is cleaner, even though there will be a slight flash on first workspace open.
Though I am open to change this in the future. Just not as part of endgame.